content/en/docs/tasks/administer-cluster/verify-signed-artifacts.md

```diff
@@ -41,58 +47,37 @@ for FILE in "${FILES[@]}"; do
 done
 ```
 
-Then verify the blob by using `cosign`:
+Then verify the blob using `cosign verify-blob`:
 
 ```shell
-cosign verify-blob "$BINARY" --signature "$BINARY".sig --certificate "$BINARY".cert
+cosign verify-blob "$BINARY" --signature "$BINARY".sig --certificate "$BINARY".cert --certificate-identity krel-staging@k8s-releng-prod.iam.gserviceaccount.com --certificate-oidc-issuer https://accounts.google.com
 ```
 
-cosign v1.9.0 is required to be able to use the `--certificate` flag. Please use
-`--cert` for older versions of cosign.
+A succesfull verification will output `Verified OK`.
 
 {{< note >}}
-To learn more about keyless signing, please refer to [Keyless
-Signatures](https://github.com/sigstore/cosign/blob/main/KEYLESS.md#keyless-signatures).
+Verification now requires identity flags, `--certificate-identity` and `--certificate-oidc-issuer`.
+
+For additional information please read the Cosign 2.0 [blog post](https://blog.sigstore.dev/cosign-2-0-released/).
 {{< /note >}}
 
 ## Verifying image signatures
 
 For a complete list of images that are signed please refer
 to [Releases](/releases/download/).
 
-Let's pick one image from this list and verify its signature using
-the `cosign verify` command:
+Pick one image from this list and verify its signature using `cosign verify`:
 
 ```shell
-COSIGN_EXPERIMENTAL=1 cosign verify registry.k8s.io/kube-apiserver-amd64:v{{< skew currentVersion >}}.0
+cosign verify registry.k8s.io/kube-apiserver-amd64:v{{< skew currentVersion >}}.0 --certificate-identity krel-trust@k8s-releng-prod.iam.gserviceaccount.com --certificate-oidc-issuer https://accounts.google.com
 ```
 
-{{< note >}}
-`COSIGN_EXPERIMENTAL=1` is used to allow verification of images signed
-in `KEYLESS` mode. To learn more about keyless signing, please refer to
-[Keyless Signatures](https://github.com/sigstore/cosign/blob/main/KEYLESS.md#keyless-signatures)
-. {{< /note >}}
-
-### Verifying images for all control plane components
```

Comment:
Why remove this?

Reply:
I've explained why I removed the entry in a [previous comment](https://github.com/kubernetes/website/pull/40910#issuecomment-1529142829). Those commands don't work as written.I tried different alternatives but none of them produced useful output

Reply:
/hold

We should find another fix. Ideally, we find a way to retain that heading and provide correct (or at least correct-for-the-alpha) advice.

It's OK to unhold this if SIG Release confirm that it's not possible for an end user to verify container image artifacts are genuine.

Reply:
There's a related concern: the PR short heading doesn't mention or hint at this removal.

Reply:
An end user CAN verify if control plane container images are genuine using `cosign`, but providing a list of image names in a prepopulated file. What I wasn't able to do is to automatically produce a list of all the control plane images using the command provided in the document:

```
curl -Ls https://sbom.k8s.io/$(curl -Ls https://dl.k8s.io/release/latest.txt)/release | grep 'PackageName: registry.k8s.io/' | awk '{print $2}' > images.txt
input=images.txt
while IFS= read -r image
do
  COSIGN_EXPERIMENTAL=1 cosign verify "$image"
done < "$input"
```

I think that the content of the registries has changed, probably this script was written when the images were stored on `k8s.gcr.io` but now on `k8s.gcr.io`, I think that images are also stored on AWS S3 buckets as well and they are probably using more complex redirects, so my guess is that those changes are breaking the functionality of this script.

As you said, getting input from the teams directly involved in these areas would be good.

ACK, if content is removed, it should be clearly stated in the PR comment.

Thanks!

Reply:
I have a working prototype shell script, it's a lot longer than the original but it works. I'm sure it can be improved but this is what I have for now:

```shell
#!/bin/bash

# Control plane image names
images=(kube-apiserver kube-controller-manager kube-scheduler kube-proxy conformance)

# Get the latest stable image version from dl.k8s.io/release/stable.txt
version=$(curl -Ls https://dl.k8s.io/release/stable.txt)

#  Image architecture name
arch=$1

# Require the image architecture name
[ $# -eq 0 ] && { echo "missing arch name [amd64|arm64|ppc64le|s390x]"; exit 1; }

# Loop through the images and verify them using cosign
for image in "${images[@]}"; do
  # Get the image names from registry.k8s.io
  name=$(curl -s https://registry.k8s.io/v2/${image}/tags/list | grep -o "\"${arch}-v[0-9.]*\"" | awk -F\" '{print $2}')
  
  # Verify each image with cosign using registry/name-arch:version --certificate-identity --certificate-oidc-issuer
  echo "===> Verifiying image: registry.k8s.io/${image}${name}-${arch}:${version} <==="
  cosign verify "registry.k8s.io/${image}${name}-${arch}:${version}" --certificate-identity krel-trust@k8s-releng-prod.iam.gserviceaccount.com --certificate-oidc-issuer https://accounts.google.com | jq -r .
done
```

This another option, which is the closest I could get to the original script, but it produces a list of images with lots of errors at the time of verifying them, like they either don't exist or the name format is incorrect, not sure:

```shell
arch="amd64"
version=$(curl -Ls https://dl.k8s.io/release/stable.txt)
curl -Ls "https://sbom.k8s.io/${version}/source" | grep 'PackageName: k8s.io/' | awk '{print $2}' > images.txt
input=images.txt
while IFS= read -r image
do
  cosign verify "registry.${image}-${arch}:${version}" --certificate-identity krel-trust@k8s-releng-prod.iam.gserviceaccount.com --certificate-oidc-issuer https://accounts.google.com | jq -r .
done < "$input"
```

Suggestions welcomed. Thanks!

FYI: @kranurag7 @sftim

Reply:
Update: I think that Rijul Gulati in PR #40982 found the right combination of `curl+grep+cut+sed` to retrieve the names of the control plane images that need to be verified with cosign. So, IMO, the winning command would be:

```
curl -Ls "https://sbom.k8s.io/$(curl -Ls https://dl.k8s.io/release/stable.txt)/release" | grep "SPDXID: SPDXRef-Package-registry.k8s.io" |  grep -v sha256 | cut -d- -f3- | sed 's/-/\//' | sed 's/-v1/:v1/' > images.txt
input=images.txt
while IFS= read -r image
do
  cosign verify "$image" --certificate-identity krel-trust@k8s-releng-prod.iam.gserviceaccount.com --certificate-oidc-issuer https://accounts.google.com
done < "$input"
```

What is missing here are the values for `--certificate-identity` and `--certificate-oidc-issuer` for image and for blob. What should we use for these.

For the verify-blob command, the right arguments are:

`cosign verify-blob "$BINARY" --signature "$BINARY".sig --certificate "$BINARY".cert --certificate-identity krel-staging@k8s-releng-prod.iam.gserviceaccount.com --certificate-oidc-issuer https://accounts.google.com`

Result is:
`Verified OK`


The variations are `--certificate-identity krel-trust@k8s-releng-prod.iam.gserviceaccount.com` for `verify` and `--certificate-identity krel-staging@k8s-releng-prod.iam.gserviceaccount.com` for `verify-blob` respectively.

If all this can be settled, I can close this PR and have Rijul proceed with his (much cleaner) solution but adding these new required options for Cosign 2.0. But I'll leave that to the experts, these are just my 0.2 cents... 

FYI @sftim @PushkarJ @dipesh-rawat

Thanks!